scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -12,6 +12,7 @@
 from scipy.optimize._constraints import (Bounds, NonlinearConstraint,
                                          LinearConstraint)
 from scipy.optimize import rosen
+from scipy.sparse import dok_matrix
```

Comment:
do the modifications work with all kinds of sparse matrices? Do we need to check that?

Reply:
I replied the same in the other thread :sweat_smile: , the changes work with all kinds of sparse matrices, I did not add all to the test to avoid cluttering them, but I can do so and it's easy if you want.

Reply:
It looks like the only thing we're doing with the sparse matrix is calling `.dot()` on it, so testing one kind is sufficient. I'd recommend using `csr_matrix` instead of `dok_matrix` for the test, though.